### DISCUSSION TITLE
404 Not Found when I try to get the data

### DISCUSSION THREAD

**Discussion by sboal94** [kreait/firebase-php#524] [Q&A] [answered]
Well, I'm trying to connect to Firebase realtime database from php in the following way:
```
use Kreait\Firebase\Factory;
require 'vendor/autoload.php';
require 'init/config.php';

$factory = (new Factory)->withServiceAccount(../firebase.json);
$database = $factory->createDatabase();
$reference = $database->getReference('/courses');
$value = $reference->getValue();
```

I have also tried with 
```
$reference = $database->getReference('courses');
$snapshot = $reference->getSnapshot();
$value = $snapshot->getValue();
```
And
```
$database->getReference('courses')->getChildKeys();
```
And everything gives the same result...
I don't know what I'm doing wrong, if anyone can give me a hand 

**Firebase rules:**
```
{
  "rules": {
        ".read": true,
        ".write": true,
  }
}
```
**Database:**
![Captura de pantalla 2020-12-11 a las 13 56 23](https://user-images.githubusercontent.com/36601938/101906400-2de67a80-3bb9-11eb-8d71-53f28eb9d26e.png)

When I try to do `<?php echo $value; ?>` the response is:

Fatal error: Uncaught GuzzleHttp\Exception\ClientException: Client error: `GET https://myWeb.firebaseio.com/courses` resulted in a `404 Not Found` response: in /Users/myName/sites/web/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113 Stack trace: #0 /Users/myName/sites/web/vendor/guzzlehttp/guzzle/src/Middleware.php(69): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response), NULL, Array, NULL) #1 /Users/myName/sites/web/vendor/guzzlehttp/promises/src/Promise.php(204): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response)) #2 /Users/myName/sites/web/vendor/guzzlehttp/promises/src/Promise.php(153): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), NULL) #3 /Users/myName/sites/web/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}() #4 /Users/myName/sites/web/vendor/guzzlehttp/promise in /Users/myName/sites/web/vendor/kreait/firebase-php/src/Firebase/Exception/DatabaseApiExceptionConverter.php on line 62

**I am working with :**
macOs Big Sur 11.0.1
MAMP (same as xampp in windows)
PHP 7.4.12 
kreait/firebase-php 5.13.0

**Output of `composer show | grep kreait`:**
kreait/clock                   1.1.0   A PHP 7.0 compatible clock abstraction
kreait/firebase-php            5.13.0  Firebase Admin SDK
kreait/firebase-tokens         1.14.0  A library to work with Firebase tokens

Thanks a lot!! 👍

**Comment by jeromegamez** [comment, marked as answer]
Thank you for the detailed description! It was said on the discord server that Firebase has changed the way it names the default Realtime Database name on new projects, and I believe (hope 😅) that this is the case here.

Could you please check on https://console.firebase.google.com/project/_/database/ what the name of your database is? You can find it right of the 🔗  symbol and it should look something like `https://<some-string>.firebaseio.com` (By default, the SDK will build this URL from the `project_id` field in your service account JSON file).

Once you have that URL, you can [configure it in the SDK with the `withDatabaseUri()` method of the `Factory` class](https://firebase-php.readthedocs.io/en/5.13.0/setup.html#custom-database-uri), so for example:

```php
$factory = (new Factory)
    ->withServiceAccount(../firebase.json)
    ->withDatabaseUri('https://my-database-name.firebaseio.com');

$database = $factory->createDatabase();
```

Please let me know if this works 🤞

**Reply by sboal94** [reply]
Thank you very much for the solution! It was just that! Awesome! 👍 🥇
